### PR TITLE
MGMTBUGSM-537: Create an utility GetIgnitionEndpoint function

### PR DESCRIPTION
**Pull request by mkowalski**
This is a manual cherry-pick of #4274 and #4289.

### CODE REVIEW

File: internal/host/hostutil/host_utils.go [hunk 1]
```diff
@@ -208,3 +210,35 @@ func SaveDiskPartitionsIsSet(installerArgs string) bool {
...
+	}
+
+	if host.MachineConfigPoolName != "" {
+		poolName = host.MachineConfigPoolName
+	}
+
+	// Specify ignition endpoint based on cluster configuration:
+	address := cluster.APIVip
+	if address == "" {
+		address = swag.StringValue(cluster.APIVipDNSName)
```
[the remark this answers is not in the document]
Comment (mkowalski) [reply internal/host/hostutil/host_utils.go:231]: Yes, this code has been refactored a lot between 2.5 and master

**Review by carbonin — APPROVED** [review]

**Comment by mkowalski** [comment]
@filanov I guess I need you to add `bugzilla/valid-bug`

**Pull request merged**